File: 2024/day_26/solution.js
function getCompleted(timeWorked, totalTime) {
  function getSeconds(str) {
    const parts = str.split(":");
    let total = 0;

    for (let i = 0; i < parts.length; i++) {
      total = total * 60 + parseInt(parts[i], 10);
    }

    return total;
  }

  const workedSeconds = getSeconds(timeWorked);
  const totalSeconds = getSeconds(totalTime);
  const percentage = Math.round((workedSeconds / totalSeconds) * 100);

  return `${percentage}%`;
}

File: 2024/day_26/solution.py
def get_completed(time_worked: str, total_time: str) -> str:
    def to_seconds(time_str: str) -> int:
        h, m, s = map(int, time_str.split(':'))
        return h * 3600 + m * 60 + s
    
    def gcd(a: int, b: int) -> int:
        while b:
            a, b = b, a % b
        return a
    
    worked_seconds = to_seconds(time_worked)
    total_seconds = to_seconds(total_time)
    
    # Calculate percentage and round to nearest int
    percentage = round((worked_seconds / total_seconds) * 100)
    
    return f"{percentage}%"
